### PR TITLE
[SMALLFIX] Clean up in the keyvalue module

### PR DESCRIPTION
**Pull request by HorizonNet**
Minor fixes in the **keyvalue** module.

*BaseKeyValueStoreReader*: Introduced automatic resource management in try statement
*KeyValueOutputCommitter*: Fixed deprecation

### CODE REVIEW

File: keyvalue/client/src/main/java/alluxio/hadoop/mapreduce/KeyValueOutputCommitter.java [hunk 1]
```diff
@@ -109,6 +109,6 @@ public void abortTask(TaskAttemptContext taskContext) {
    * @return the temp directory name
    */
   public static String getPendingDirName() {
-    return FileOutputCommitter.TEMP_DIR_NAME;
+    return FileOutputCommitter.PENDING_DIR_NAME;
```
Comment (aaudiber) [inline keyvalue/client/src/main/java/alluxio/hadoop/mapreduce/KeyValueOutputCommitter.java]: Unfortunately we're stuck with the deprecated constant as long as we support hadoop 1: https://hadoop.apache.org/docs/r1.2.1/api/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.html

**Comment by aaudiber** [comment]
@HorizonNet I left one comment on the code, looks good otherwise

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comment. I changed it back and left a comment.

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**